### PR TITLE
Update case implementation

### PR DESCRIPTION
**Pull request by AsterNighT**
Signed-off-by: AsterNighT <klxjt99@outlook.com>

### CODE REVIEW

File: pkg/cases/handler.go [hunk 1]
```diff
@@ -94,9 +123,18 @@ func (h *CaseHandler) DeleteCaseByCaseID(c echo.Context) error {
 // @Success 200 {object} api.ReturnedData{data=[]Case}
 // @Router /patient/{patientID}/case/{caseID} [GET]
 func (h *CaseHandler) GetPreviousCases(c echo.Context) error {
-	// ...
+	db := GetDB()
+	var case1 Case
+	var cases []Case
+	db.First(&case1, c.Param("caseID"))
+	cases = append(cases, case1)
```
Comment (RupertPaoZ) [inline pkg/cases/handler.go]: i don't think we need to return current case

**Review by RupertPaoZ — DISMISSED** [review]

File: pkg/cases/handler.go [hunk 2]
```diff
@@ -37,9 +54,14 @@ func (h *CaseHandler) GetAllCases(c echo.Context) error {
 // @Success 200 {object} api.ReturnedData{data=Case}
 // @Router /patient/{patientID}/case [GET]
 func (h *CaseHandler) GetLastCaseByPatientID(c echo.Context) error {
-	// ...
+	db := utils.GetDB()
+	db.Where("PatientID = ?", c.Param("patientID"))
```
Comment (ziyue-pan) [inline pkg/cases/handler.go:56]: Is that an elegant design to put GORM operations in the router handler?

Considering validator and jwt verification, maybe separate HTTP and GORM operations.

Comment (AsterNighT) [reply pkg/cases/handler.go:56]: I think it's ok to do so? In fact there's nothing to do in the handler if we put db operations elsewhere. The auth part should be done via middleware and the validation could be somewhere wraped in the `json.decoder`.

Comment (ziyue-pan) [reply pkg/cases/handler.go:56]: > I think it's ok to do so? In fact there's nothing to do in the handler if we put db operations elsewhere. The auth part should be done via middleware and the validation could be somewhere wraped in the `json.decoder`.

Are `middleware` and `json.decoder` sort of plugins to Echo?

Comment (ziyue-pan) [reply pkg/cases/handler.go:56]: by the way, is there any plan to implement the jwt validation recently.   It's quite needed in all 4 groups' designs

Comment (AsterNighT) [reply pkg/cases/handler.go:56]: Middleware is a sort of design pattern, and is provided by echo. `json.decoder` is simply the process of extracting data from requests. For now you could simply ignore the jwt validation, or ask for an API for that. The detailed implementation does not affect us.

**Comment by AsterNighT** [comment]
@RupertPaoZ PTAL

**Review by RupertPaoZ — APPROVED** [review]

File: pkg/cases/model.go [hunk 3]
```diff
@@ -5,39 +5,39 @@ import (
 )
 
 type Case struct {
-	ID             uint `gorm:"primaryKey"` // Every object should have ID
-	PatientID      uint // A has many relationship should be on this
+	ID             uint `gorm:"primaryKey"`   // Every object should have ID
+	PatientID      uint `validate:"required"` // A has many relationship should be on this
 	DoctorID       uint
```
Comment (ziyue-pan) [inline pkg/cases/model.go:10]: These seem common attributes, rather than reference keys. Does this meet your expectation?

Comment (AsterNighT) [reply pkg/cases/model.go:10]: This is somehow required by `gorm` `hasMany` relationship.

**Pull request merged**